src/sentry/workflow_engine/models/detector.py

```diff
@@ -23,15 +23,27 @@
 class Detector(DefaultFieldsModel, OwnerModel):
     __relocation_scope__ = RelocationScope.Organization
 
-    organization = FlexibleForeignKey("sentry.Organization")
+    # TODO - Finish removing this field
+    organization = FlexibleForeignKey("sentry.Organization", on_delete=models.CASCADE, null=True)
+
+    project = FlexibleForeignKey("sentry.Project", on_delete=models.CASCADE, null=True)
     name = models.CharField(max_length=200)
 
     # The data sources that the detector is watching
     data_sources = models.ManyToManyField(
         "workflow_engine.DataSource", through="workflow_engine.DataSourceDetector"
     )
 
-    # The conditions that must be met for the detector to be considered 'active'
+    # If the detector is not enabled, it will not be evaluated. This is how we "snooze" a detector
+    enabled = models.BooleanField(default=True)
+
```

Comment:
Weren't we going to add a `created_by` field to this model to preserve the historical data?

Reply:
👍 i couldn't remember if it was this or just the workflow that needed it. will update.

Reply:
added the `created_by` -- for the `threshold_period` we'll use the `config` field, and a CONFIG_SCHEMA to define the attributes there. this will need to be part of the `MetricDetector`